### PR TITLE
Expanded the supported models list per tests

### PR DESCRIPTION
**Pull request by taylanbil**
Tested all the TV models. All of them work, except for squeezenet. I'm going to look at why that failed next.

I added them to the supported models list in the imagenet train script.

I had to change some parts for inception as its preprocessing requires
special handling.

I trained on 64 vCPUs, 240 GB memory machine (200g shm w/ docker), 8 tpu cores 

```bash
#!/bin/bash

image="gcr.io/tpu-pytorch/xla:nightly"
docker_args="--shm-size 200G -e XRT_TPU_CONFIG=\"tpu_worker;0;$GREENVMIP:8470\""
docker_args="$docker_args --name inceptionv3 --rm=true "

pytorch_cmd="python $code_source/test/test_train_imagenet.py \
  --datadir='$data_source' --model='$1' --num_epochs=1 --metrics_debug"

ssh -i ~/.ssh/google_compute_engine taylanbil@$REDVMIP -t \
  "docker run -it $docker_args $image $pytorch_cmd"
```

### CODE REVIEW

**Comment by asuhan** [comment]
Cool, which version of `torchvision` is this? There are several new models under https://github.com/pytorch/vision/tree/master/torchvision/models, plus the detection (Mask R-CNN) and segmentation models.

**Comment by taylanbil** [comment]
@asuhan sorry, I should  have been more precise; these are "most" of the *classification* models as seen in: https://pytorch.org/docs/stable/torchvision/models.html. I see now that I'm missing a few classification models. Will add them too. Thanks.

File: test/test_train_imagenet.py [hunk 1]
```diff
@@ -1,7 +1,50 @@
 import test_utils
 
+ALEXNET = 'alexnet'
```
Comment (dlibenzi) [inline test/test_train_imagenet.py]: I think, since those have only one use, we can just inline the strings within the dictionary.
Removes quite a bit of code.

File: test/test_train_imagenet.py [hunk 2]
```diff
@@ -50,15 +94,25 @@
     setattr(FLAGS, arg, value)
 
 
+def load_torchvision_model(model_name):
```
Comment (dlibenzi) [inline test/test_train_imagenet.py]: We have a couple of places where we do `if inception`.
Maybe we can have something like this?
```Python
MODEL_PROPERTIES = {
  'inception': {
    img_dim: 299,
    model_fn: lambda: torchvision.models.inception_v3(aux_logits=False),
  },
  'DEFAULT': {
    img_dim: 224,
    model_fn: getattr(torchvision.models, FLAGS.model),
  }
}

def get_model_property(name):
  pdict = MODEL_PROPERTIES.get(FLAGS.model, None) or MODEL_PROPERTIES['DEFAULT']
  return pdict[name]
```

**Comment by asuhan** [comment]
Some of these models might use the CPU interop for flavors of convolution we don't support, would be nice to retrieve the counters with `--metrics_debug` and file issues for each of them. There shouldn't be more than a handful at most, probably just convolution with dilation.

**Comment by dlibenzi** [comment]
@taylanbil Did you pass this through `google-pyformat`?

**Comment by taylanbil** [comment]
@asuhan these models are the ones available in anaconda; see https://github.com/pytorch/xla/blob/master/scripts/build_torch_wheels.sh#L55

googlenet and such aren't available in the doker img I'm working with, I'm working on adding them though.

**Comment by taylanbil** [comment]
resnet50 works. Trying squeezenet.

**Comment by taylanbil** [comment]
I just fired up a script that'll run everything from scratch on tpu. I'm leaving now, will report the results.

**Comment by dlibenzi** [comment]
Taylan, LMK if this is OK to merge ....

**Comment by taylanbil** [comment]
Checking the logs, densenet, inception, alexnet, resnets finished normally.

squeezenet and vgg runs errored due to network. Re-running them. I will let you know about the results.

**Comment by taylanbil** [comment]
@dlibenzi this should be safe to merge since squeezenet is commented out. vgg's are running smoothly as well.

**Comment by dlibenzi** [comment]
> Checking the logs, densenet, inception, alexnet, resnets finished normally.
> 
> squeezenet and vgg runs errored due to network. Re-running them. I will let you know about the results.

What was the issue with squeezenet?
Sorry ... short memory ... :grin:

**Comment by taylanbil** [comment]
@dlibenzi I'm not sure. error logs are in the drive. It seems to be failing when trying to maxpool.
